torch/utils/data/distributed.py

```diff
@@ -65,3 +65,117 @@ def __len__(self):
 
     def set_epoch(self, epoch):
         self.epoch = epoch
+
+class ChunkDataReader(object):
+    r"""Reads a chunk of data given a chunk index.
+
+    A chunk could be full file, section of a large file, folders, URLs, or any other
+    abstraction that allows data to be segmented roughtly the same size.
+
+    As an example, chunking could be used in a scenario where MNIST dataset for training 
+    would be converted in CSV text files and equally splitted in several files,
```

Comment:
As an example, chunking could be used in a scenario where a dataset is split between several csv files each containing complete records or examples. In this case, each file could be a chunk. Similarly a large csv file can be considered as a collection of chunks by defining chunk boundary based on some physical size (e.g. 32MB) and seeking to the first record in each chunk during reading.

Reply:
Let's not use a specific example like MNIST here.